react/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,163 @@
+
+  const onDropImage = async (files: any[]) => {
+    setError(null)
+
+    try {
+      if (files?.[0]) {
+        setIsLoading(true)
+
+        const { data, errors } = await uploadFile({
+          variables: { file: files[0] },
+        })
+
+        if (errors) {
+          setError(
+            intl.formatMessage({
+              id: 'store/form.document-uploader.error.file-size',
+            })
+          )
+          return
+        }
+
+        setIsLoading(false)
+        setIsOpen(false)
+        setFileName(files?.[0].name)
+        setImageUrl(data?.uploadFile.fileUrl)
+
+        return data
+      }
+      return setError(
+        intl.formatMessage({
+          id: 'store/form.document-uploader.error.file-size',
+        })
+      )
+    } catch (err) {
+      setError(
+        intl.formatMessage({ id: 'store/form.document-uploader.error.generic' })
+      )
+      setIsLoading(false)
+    }
+  }
+
+  const { getInputProps, getRootProps } = useDropzone({
+    accept: '.pdf, image/*',
+    maxSize: MAX_SIZE,
+    multiple: false,
+    onDrop: onDropImage,
+  })
+
+  const handleAddImage = () => {
+    setIsOpen(false)
+  }
+
+  return (
+    <div className="vtex-styleguide-9-x-dropdown vtex-dropdown" ref={ref}>
+      <span className="db mb3 w-100 c-on-base t-small">
+        Operating Agreement
+      </span>
+      <StyleButton
+        title={intl.formatMessage({ id: 'store/form.add-document.title' })}
+        active={isOpen}
+        onToggle={() => setIsOpen(!isOpen)}
+        style={null}
+        label={
+          <div className="flex flex-row justify-between items-center w-100 pa4">
+            <IconImage />
+            <IconCaretDown size={8} />
+          </div>
+        }
+      />
+
+      {isOpen && (
+        <div className="absolute pa5 bg-white b--solid b--muted-4 bw1 br2 w5 z-1">
+          {isLoading && (
+            <div className="absolute flex justify-center items-center top-0 left-0 h-100 w-100 br2 z-1 bg-black-05">
+              <Spinner />
+            </div>
+          )}
+
+          <div className={`flex flex-column ${isLoading && 'o-20'}`}>
+            <div className="mb4">
+              <Input
+                label={intl.formatMessage({
+                  id: 'store/form.add-document.label',
+                })}
+                onChange={(e: any) => {
+                  setImageUrl(e.target.value)
+                }}
+                placeholder={intl.formatMessage({
+                  id: 'store/form.add-document.placeholder',
+                })}
+              />
+            </div>
+
+            <Button onClick={handleAddImage} size="small" disabled={!imageUrl}>
+              <FormattedMessage id="store/form.add-button" />
+            </Button>
+
+            <div className="flex flex-column">
+              <span className="db mb3 w-100 c-on-base t-small">
+                <FormattedMessage id="store/form.upload-document-label" />
```

Comment:
Change it to `IOMessage` after you define the messages.

Reply:
Done!